### PR TITLE
Validate blockhash that the relayer responds with

### PR DESCRIPTION
**Pull request by avalonche**
## 📝 Summary

Validates the block hash the relay responds with the hash calculated from the relay block.

## ⛱ Motivation and Context

Resolves https://github.com/flashbots/mev-boost/issues/280
## 📚 References

---

## ✅ I have run these commands

* [x] `make lint`
* [x] `make test-race`
* [x] `go mod tidy`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,24 +4,40 @@ go 1.18
 
 require (
 	github.com/ethereum/go-ethereum v1.10.17
-	github.com/flashbots/go-boost-utils v0.3.5
+	github.com/flashbots/go-boost-utils v1.0.1-0.20220908121238-1df8f336bae3
```
Comment (metachris) [inline go.mod:7]: v1.0.0 should have everything?

File: server/service_test.go [hunk 2]
```diff
@@ -506,6 +525,23 @@ func TestGetPayload(t *testing.T) {
...
+	// 		BlockNumber:  12346,
+	// 		FeeRecipient: payload.Message.Body.ExecutionPayloadHeader.FeeRecipient,
+	// 	}
+
+	// 	backend.relays[0].GetPayloadResponse = resp
+	// 	rr := backend.request(t, http.MethodPost, path, payload)
+	// 	require.Equal(t, http.StatusBadGateway, rr.Code, rr.Body.String())
+	// 	require.Equal(t, 1, backend.relays[0].GetRequestCount(path))
+	// 	require.Equal(t, `{"code":502,"message":"no successful relay response"}`+"\n", rr.Body.String())
+	// })
```
Comment (jtraglia) [inline server/service_test.go:544]: Please delete the commented out test.

```suggestion
```

**Review by jtraglia — CHANGES_REQUESTED** [review]
Looks like a good check to have 👍 one small suggestion though.

**Review by metachris — APPROVED** [review]

File: server/service_test.go (hunk 2 above)
Comment (metachris) [reply server/service_test.go]: nice, committed

**Review by jtraglia — APPROVED** [review]
LGTM 👍

**Pull request merged**